internal/jennies/java/rawtypes.go

```diff
@@ -0,0 +1,278 @@
+package java
+
+import (
+	"fmt"
+	"github.com/grafana/codejen"
+	"github.com/grafana/cog/internal/ast"
+	"github.com/grafana/cog/internal/jennies/common"
+	"github.com/grafana/cog/internal/tools"
+	"path/filepath"
+	"strings"
+)
+
+type RawTypes struct {
+	config  Config
+	imports *common.DirectImportMap
+
+	typeFormatter *typeFormatter
+}
+
+func (jenny RawTypes) JennyName() string {
+	return "JavaBuilder"
+}
+
+func (jenny RawTypes) Generate(context common.Context) (codejen.Files, error) {
+	files := make(codejen.Files, 0)
+	jenny.imports = NewImportMap()
+	jenny.typeFormatter = createFormatter(context)
+
+	for _, schema := range context.Schemas {
+		output, err := jenny.genFilesForSchema(schema)
+		if err != nil {
+			return nil, err
+		}
+
+		files = append(files, output...)
+	}
+
+	return files, nil
+}
+
+func (jenny RawTypes) genFilesForSchema(schema *ast.Schema) (codejen.Files, error) {
+	files := make(codejen.Files, 0)
+	scalars := make(map[string]ast.ScalarType)
+
+	packageMapper := func(pkg string, class string) string {
+		if pkg == schema.Package {
+			return ""
+		}
+
+		return jenny.imports.Add(class, pkg)
+	}
+
+	jenny.typeFormatter = jenny.typeFormatter.withPackageMapper(packageMapper)
+
+	for _, object := range schema.Objects {
+		jenny.imports = NewImportMap()
+		if object.Type.IsMap() {
+			continue
+		}
+		if object.Type.IsScalar() {
+			scalars[object.Name] = object.Type.AsScalar()
+			continue
+		}
+
+		output, err := jenny.generateSchema(schema.Package, object)
+		if err != nil {
+			return nil, err
+		}
+
+		filename := filepath.Join(
+			strings.ToLower(schema.Package),
+			fmt.Sprintf("%s.java", tools.UpperCamelCase(object.Name)),
+		)
+
+		files = append(files, *codejen.NewFile(filename, output, jenny))
+	}
+
+	if len(scalars) > 0 {
+		output, err := jenny.formatScalars(schema.Package, scalars)
+		if err != nil {
+			return nil, err
+		}
+
+		filename := filepath.Join(strings.ToLower(schema.Package), "Constants.java")
+		files = append(files, *codejen.NewFile(filename, output, jenny))
+	}
+
+	return files, nil
+}
+
+func (jenny RawTypes) generateSchema(pkg string, object ast.Object) ([]byte, error) {
+	switch object.Type.Kind {
+	case ast.KindStruct:
+		return jenny.formatStruct(pkg, object)
+	case ast.KindEnum:
+		return jenny.formatEnum(pkg, object)
+	case ast.KindRef:
+		return jenny.formatReference(pkg, object)
+	case ast.KindIntersection:
+		return jenny.formatIntersection(pkg, object)
+	}
+
+	return nil, nil
+}
+
+func (jenny RawTypes) formatEnum(pkg string, object ast.Object) ([]byte, error) {
+	var buffer strings.Builder
+
+	enum := object.Type.AsEnum()
+	values := make([]EnumValue, len(enum.Values))
+	for i, value := range enum.Values {
+		values[i] = EnumValue{
+			Name:  strings.ToUpper(value.Name),
+			Value: value.Value,
+		}
+	}
+
+	enumType := "Integer"
```

Comment:
Probably it isn't 100% accurate. 

In [`cue`](https://github.com/grafana/cog/blob/main/internal/simplecue/generator.go#L212) we are setting `ast.KindInt64`. Its a `Long` in Java and it forces to add casting to the values in the generated code. So at the moment using `Integer` makes it simpler and works at the moment.